Source/ARTDefault.m

```diff
@@ -6,9 +6,35 @@
 //  Copyright (c) 2015 Ably. All rights reserved.
 //
 
-#import "ARTDefault+Private.h"
-
 #import "Ably.h"
+#import "ARTDefault+Private.h"
+#import <sys/utsname.h>
+
+// NSOperatingSystemVersion has NSInteger as version components for some reason, so mitigate it here.
+static UInt32 conformVersionComponent(const NSInteger component) {
+    if (component < 0) return 0;
+    UInt32 conformedValue = (UInt32)component;
+    return conformedValue;
+}
+
+static NSString *canonizeStringAsUserAgentToken(NSString *_Nonnull inputString) {
```

Comment:
This method is looking complex enough that it should have some unit tests.

Reply:
https://github.com/ably/ably-cocoa/pull/1119/commits/8e0203c2f37cccdee60b1e20ea8082386a1ece02